src/main/java/org/spongepowered/api/util/command/spec/CommandSpec.java

```diff
@@ -0,0 +1,392 @@
+     * @param args The arguments to process with
+     * @param context The context to put data in
+     * @throws ArgumentParseException if an invalid argument is provided
+     */
+    public void populateContext(CommandSource source, CommandArgs args, CommandContext context) throws ArgumentParseException {
+        this.args.parse(source, args, context);
+        if (args.hasNext()) {
+            args.next();
+            throw args.createError(t("Too many arguments!"));
+        }
+    }
+
+    /**
+     * Return tab completion results using the existing parsed arguments and context. Primarily useful when including a subcommand in an existing
+     * specification.
+     *
+     * @param source The source to parse arguments for
+     * @param args The arguments object
+     * @param context The context object
+     * @return possible completions, or an empty list if none
+     */
+    public List<String> complete(CommandSource source, CommandArgs args, CommandContext context) {
+        Preconditions.checkNotNull(source, "source");
+        List<String> ret = this.args.complete(source, args, context);
+        return ret == null ? ImmutableList.<String>of() : ImmutableList.copyOf(ret);
+    }
+
+    /**
+     * Get the active executor for this command. Generally not a good idea to call this directly,
+     * unless you are handling arg parsing specially
+     *
+     * @return The active executor for this command
+     */
+    public CommandExecutor getExecutor() {
+        return this.executor;
+    }
+
+    /**
+     * Get the active input tokenizer used for this commmand.
+     *
+     * @return This command's input tokenizer
+     */
+    public InputTokenizer getInputTokenizer() {
+        return this.argumentParser;
+    }
+
+
+    @Override
+    public Optional<CommandResult> process(CommandSource source, String arguments) throws CommandException {
+        checkPermission(source);
+        final CommandArgs args = new CommandArgs(arguments, getInputTokenizer().tokenize(arguments, false));
+        final CommandContext context = new CommandContext();
+        this.populateContext(source, args, context);
+        return Optional.of(getExecutor().execute(source, context));
+    }
+
+    @Override
+    public List<String> getSuggestions(CommandSource source, String arguments) throws CommandException {
+        CommandArgs args = new CommandArgs(arguments, getInputTokenizer().tokenize(arguments, true));
+        return complete(source, args, new CommandContext());
+    }
+
+    @Override
+    public boolean testPermission(CommandSource source) {
+        return this.permission == null || source.hasPermission(this.permission);
+    }
+
+    /**
+     * Get a short, one-line description used with this command if any is present
+     *
+     * @return the short description.
+     */
+    @Override
+    public Optional<Text> getShortDescription(CommandSource source) {
+        return Optional.fromNullable(this.description);
```

Comment:
You can just change the field to an `Optional` directly, no need to wrap this every time